cmd/minikube/cmd/delete.go

```diff
@@ -113,13 +113,16 @@ func runDelete(cmd *cobra.Command, args []string) {
 		if profileFlag != constants.DefaultMachineName {
 			exit.UsageT("usage: minikube delete --all")
 		}
-
-		err := oci.DeleteAllVolumesByLabel(oci.Docker, fmt.Sprintf("%s=%s", oci.CreatedByLabelKey, "=true"))
-		if err != nil { // if there is no volume there won't be any error
-			glog.Warningf("error deleting left docker volumes. \n%v:\nfor more information please refer to docker documentation: https://docs.docker.com/engine/reference/commandline/volume_prune/", err)
+		errs := oci.DeleteAllContainersByLabel(oci.Docker, fmt.Sprintf("%s=%s", oci.CreatedByLabelKey, "true"))
+		if errs != nil { // it will error if there is no container to delete
+			glog.Infof("no left over minikube  container found. %v", errs)
```

Comment:
This seems to be making a large assumption about the type of error without making it part of the API contract. How about:

`glog.Infof("delete %q containers: %v (may be ok)"), oci.CreatedByLabelKey, err)`

Please do this elsewhere instead of hardcoding "no left over" in your other error messages, to avoid the possibility of some very confusing error messages.

In the future, please consider making the oci API idempotent. It is generally not helpful to return errors for non-error conditions like this. When it comes to automating systems, any API that is not idempotent will result in fragile failure handling.

Reply:
done